temoa/model_checking/unit_checking/entry_checker.py

```
@@ -0,0 +1,64 @@
+        return True, units
+    except UndefinedUnitError:
+        return False, None
+
+
+def gather_from_table(conn: sqlite3.Connection, table: str) -> dict[str, list[int]]:
+    """gather all unique "units" entries from a table and collect the row indices"""
+
+    res = defaultdict(list)
+    with conn:
+        cur = conn.cursor()
+        cur.execute(f'SELECT units FROM {table}')
```

Ruff (0.14.8)

Possible SQL injection vector through string-based query construction

(S608)